moveit_core/package.xml

```diff
@@ -39,7 +39,8 @@
   <depend>moveit_msgs</depend>
   <depend>octomap</depend>
   <depend>octomap_msgs</depend>
-  <depend>pybind11_catkin</depend>
+  <depend condition="$ROS_DISTRO == melodic">pybind11_catkin</depend>
+  <depend condition="$ROS_DISTRO != melodic">pybind11-dev</depend>
```

Comment:
```suggestion
  <depend condition="($ROS_DISTRO == melodic) or ($ROS_DISTRO == noetic)">pybind11-dev</depend>
  <depend condition="($ROS_DISTRO != melodic) or ($ROS_DISTRO != noetic)">pybind11-dev</depend>
```

I vote for sticking with pybind11_catkin in noetic

Reply:
Once `pybind11_catkin` is independent on top of a system `pybind11-dev` package this can be simplified again.